source/decryptor/padgen.c

```diff
@@ -130,15 +130,43 @@ u32 SdPadgen()
 
 u32 NandPadgen()
 {
-    //Just trying to make sure the data we're looking for is where we think it is.
-    //Should use some sort of memory scanning for it instead of hardcoding the address, though.
-    if (*(u32*)0x080D794C != 0x5C980)
+    //Use memory know to start scanning.
+    //If not in t Should use some sort of memory scanning for it instead of hardcoding the address, though.
+    u32 ctrStart=0, c;
+    u32 listCtrStart[] = {0x080D7CAC, 0x080D748C, 0x080D740C, 0x080D794C};
+    char* listSystem[] = {"4.x", "6.x", "7.x", "9.x"};
+    u32 lenListCtrStart = sizeof(listCtrStart)/sizeof(u32);
+    for(c=0; c < lenListCtrStart; c++)
```

Comment:
We use C99, so you can declare `c` in the for loop itself if you're not using it outside of the for loop.